spock-specs/src/test/groovy/org/spockframework/docs/interaction/DetachedMockFactoryDocSpec.groovy

```diff
@@ -0,0 +1,149 @@
+package org.spockframework.docs.interaction
+
+import org.spockframework.mock.IDefaultResponse
+import org.spockframework.mock.IMockInvocation
+import org.spockframework.mock.MockUtil
+import org.spockframework.mock.ZeroOrNullResponse
+import spock.lang.Specification
+import spock.lang.Unroll
+import spock.mock.AutoAttach
+import spock.mock.DetachedMockFactory
+
+import static org.spockframework.docs.interaction.DetachedMockFactoryDocSpec.EngineMockCreator.StartMode.*
+
+class DetachedMockFactoryDocSpec extends Specification {
+  // tag::declare-static[]
+  static mockFactory = new DetachedMockFactory()
+  static mockUtil = new MockUtil()
```

Comment:
Why did you chose `static` over `@Shared`? I general I'd prefer `@Shared` over `static` whenever possible.

Reply:
I even refactored it that way the first time, before I force-pushed the static version again. The reason is simple: The inner class `EngineMockCreator` uses the same mock factory, therefore it has to be static. Would you prefer me to make the fields shared and define a redundant mock factory inside the mock creator class?